test/integration/builder/deletes.js

```diff
@@ -82,5 +82,86 @@ module.exports = function (knex) {
           );
         });
     });
+    it('#4152 Should allow returns with deletes on tables with triggers', async function () {
+      if (knex.client.driverName !== 'mssql') {
+        return true;
+      }
+
+      return await knex.transaction(async function () {
```

Comment:
I'm always wary of returning such large promises because you might start swallowing errors should it ever resolve early for some reason.
Safer approach would be something along the lines
```
let reachedEnd = false
await transaction // sets reachedEnd in the end
expect(reachedEnd).to.be(true)
```